src/module.c

```diff
@@ -11763,6 +11791,20 @@ static uint64_t moduleEventVersions[] = {
  *
  *         ValkeyModuleKey *key;    // Key name
  *
+ * * ValkeyModuleEvent_AuthenticationAttempt
+ *
+ *     Called when an authentication attempt is made, either successful or not.
+ *
+ *     The data pointer can be casted to a ValkeyModuleAuthenticationInfo
+ *     structure with the following fields:
+ *
+ *         uint64_t client_id;      // Client ID.
+ *         const char *username;    // Username used for authentication.
+ *         const char *module_name; // Name of the module that is handling the
+ *                                  // authentication. It is NULL if the
+ *                                  // authentication is handled by the core.
+ *         ValkeyModuleAuthenticationResult result;   // Result of the authentication.
```

Comment:
This documentation is so complete that it can be used without looking at the structs in valkeymodule.h, except for the result values. Let's add them too.

```suggestion
 *         ValkeyModuleAuthenticationResult result;
 *                                  // Result of the authentication:
 *                                  // VALKEYMODULE_AUTH_RESULT_GRANTED or
 *                                  // VALKEYMODULE_AUTH_RESULT_DENIED.
```